src/main/java/seedu/weeblingo/storage/LocalDatabasePopulator.java

```diff
@@ -38,18 +41,48 @@
         return flashcards;
     }
 
+    /**
+     * Gets the specified number of flashcards, as an array of flashcards.
+     *
+     * @return An array of flashcards of specified size extracted from database.
+     */
+    public static Flashcard[] getSubsetOfFlashcards(int numberOfQuestions) {
+        return Arrays.copyOfRange(
+                getDatabaseOfFlashcards(JsonDatabaseReader
+                        .readDatabaseAsJsonArray()), 0, numberOfQuestions);
+    }
+
     public static ReadOnlyFlashcardBook getDatabaseOfWeeblingo() {
         FlashcardBook sampleFb = new FlashcardBook();
-        for (Flashcard sampleFlashcard : getDatabaseOfFlashcards()) {
+        for (Flashcard sampleFlashcard : getDatabaseOfFlashcards(JsonDatabaseReader.readDatabaseAsJsonArray())) {
             sampleFb.addFlashcard(sampleFlashcard);
         }
+
+        try {
+            sampleFb.addScore(Score.of(100, 100));
+            TimeUnit.SECONDS.sleep(1);
```

Comment:
Is there a reason for calling sleep?

Reply:
This is the dummy history data generation part. Since (now the `Score` handles the `LocalDatetime` which is the `Score's` creation time internally) && (The precision of `LocalDatetime` is only limited to seconds in our application), a consecutive creation of `Score` object will result in duplicate `Score` objects stored (which, by the same principle of how we store `Flashcard`s, will throw an exception).

When the data is no longer dummy, we can safely assume the user would at least take 1 second between finishing the current quiz and finishing the next quiz.

If the above is not a legitimate assumption, I can change my implementations (but some efforts are required).

Reply:
I think it is a valid assumption! Probably will be better to document it down in the javadoc later.